exercises/sam_jenkins/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,59 @@
+#include "Shapes.h"
+#include "ShapeSorter.h"
+#include <vector>
+#include <algorithm>
+#include <iostream>
+#include <iomanip>
+using std::vector; using std::string; using std::cout;
+using std::endl; using std::sort; using std::setw;
+
+
+bool compareArea(const Shapes* s1, const Shapes *s2)
+{
+	return(s1->calcArea() > s2->calcArea());
+}
+
+bool comparePerimiter(const Shapes* s1, const Shapes *s2)
+{
+	return(s1->calcPerimiter() > s2->calcPerimiter());
+}
+
+void ShapeSorter::typePrint(string type, vector<Shapes*> shapes)
+{
+	for (vector<Shapes*>::size_type i = 0; i!= shapes.size(); ++i){
+		if (shapes[i]->getType() == type) {
+			cout << " Type: " << setw(9)<<shapes[i]->getType() << " Sides: " << setw(4) << shapes[i]->getSides() << " Area: " << setw(9) << shapes[i]->calcArea() << " Perimiter: " << setw(9) << shapes[i]->calcPerimiter() << endl;
```

Comment:
These lines are way, way to long. Just use multiple calls to cout on multiple lines to make it more readable.